lib/chorde/clients/memcached.py

```diff
@@ -44,19 +47,104 @@
     json = None  # lint:ok
 JSON_SEPARATORS = (',',':')
 
+class BasePickler(object):
+    """ Serialization/Deserialization strategy encapsulation
+
+    A pickler, in contrast with how Python's picklers work, just encapsulates
+    3 methods to serialize and deserialize: dump, dumps, and loads.
+
+    Pickler objects should contain no state as they will be long lived,
+    but they may contain immutable configuration.
+    """
+    def __init__(self, **kw):
+        self.__dict__.update(kw)
+
+    def dumps(self, obj):
+        """ Serialize the object obj into a byte string """
+        raise NotImplementedError("%s class must implement dumps", type(self).__name__)
+
+    def loads(self, s):
+        """ Read an object from the byte string s and return it """
+        raise NotImplementedError("%s class must implement loads", type(self).__name__)
+
+    def dump(self, obj, f):
+        """ Serialize the object obj into the binary file-like object f """
+        f.write(self.dumps(obj))
+
+class KeyPickler(BasePickler):
+    """ Serialization/Deserialization strategy meant for keys
+
+    Key serialization is limited to never returning control characters in their
+    serialized output, because they're forbidden for memcache keys.
+    """
+
+class ValuePickler(BasePickler):
+    """ Serialization/Deserialization strategy meant for values
+
+    Value serialization must handle large dump calls efficiently and have
+    otherwise no constraints.
+    """
+
 try:
     try:
         import ujson as cjson_
     except ImportError:
         import cjson as cjson_  # lint:ok
-    class cjson:  # lint:ok
-        loads = cjson_.decode
-        dumps = staticmethod(lambda x, separators=None, encode = cjson_.encode : encode(x))
+
+    class JSONValuePickler(ValuePickler):  # lint:ok
+        def loads(self, s, loads=cjson_.decode):
+            return loads(s)
+
+        def dumps(self, obj, dumps=cjson_.encode):
+            return dumps(obj)
+
 except ImportError:
     try:
-        import simplejson as cjson  # lint:ok
+        import simplejson as cjson_  # lint:ok
     except ImportError:
-        import json as cjson  # lint:ok
+        import json as cjson_  # lint:ok
+
+    class JSONValuePickler(ValuePickler):  # lint:ok
+        def loads(self, s, loads=cjson_.loads):
+            return loads(s)
+
+        def dumps(self, obj, dumps=cjson_.dumps):
+            return dumps(obj).encode('ascii')
+
+class JSONTextEncoder(json.JSONEncoder):
```

Comment:
`json` can be `None` from this line:

https://github.com/klaussfreire/chorde/blob/851a5f441f4d5b015b56b351057db4c85cf6c641/lib/chorde/clients/memcached.py#L47

Maybe remove that line since it's going to raise an exception anyway?

Reply:
Honestly, that's only possible if there's no json implementation, and that's unlikely since it's been part of the stdlib forever. I think that code dates back to python 2.4 days.